providers/Providers.tsx

```diff
@@ -43,23 +43,7 @@ export function Providers({ children }: { children: ReactNode }) {
   //   }
   // }, []);
 
-  const endpoint = useMemo(() => {
-    switch (env) {
-      case 'mainnet':
-        return process.env.NEXT_PUBLIC_MAINNET_RPC_URL;
-      case 'eclipse-mainnet':
-        return process.env.NEXT_PUBLIC_ECLIPSE_MAINNET_RPC_URL;
-      case 'sonic-devnet':
-        return process.env.NEXT_PUBLIC_SONIC_DEVNET_RPC_URL;
-      case 'eclipse-devnet':
-        return process.env.NEXT_PUBLIC_ECLIPSE_DEVNET_RPC_URL;
-      case 'localhost':
-        return 'http://localhost:8899';
-      case 'devnet':
-      default:
-        return process.env.NEXT_PUBLIC_DEVNET_RPC_URL;
-    }
-  }, [env]);
+  const endpoint = useMemo(() => envOptions.find(({ env: e }) => e === env)?.endpoint, [env]);
```

Comment:
not sure if the app correctly handles `endpoint` being `null`, it should probably be defaulted to something (solana mainnet?) if the `find` does not find anything